src/components/SelectInput/SelectInput.test.jsx

```diff
@@ -0,0 +1,284 @@
+          onFocus={mockedHandleFocus}
+          placeholder="Test Placeholder"
+          options={selectOptions}
+        />,
+      );
+
+      fireEvent.focus(screen.getByRole('textbox'));
+
+      expect(mockedHandleFocus).toBeCalledTimes(1);
+    });
+
+    test('it fires onBlur callback on blur', () => {
+      const mockedHandleBlur = jest.fn();
+
+      render(
+        <SelectInput
+          id="testId"
+          onBlur={mockedHandleBlur}
+          placeholder="Test Placeholder"
+          options={selectOptions}
+        />,
+      );
+
+      fireEvent.blur(screen.getByRole('textbox'));
+
+      expect(mockedHandleBlur).toBeCalledTimes(1);
+    });
+  });
+
+  describe('React-Select', () => {
+    describe('single select', () => {
+      test('it has no items selected on initial render', () => {
+        render(
+          <form data-testid="form">
+            <label htmlFor="iceCreamFlavors">Ice Cream Flavors</label>
+            <Select options={selectOptions} name="iceCreamFlavors" id="iceCreamFlavors" />
+          </form>,
+        );
+
+        expect(screen.getByTestId('form')).toHaveFormValues({ iceCreamFlavors: '' });
+      });
+
+      test('it sets the selected value for item that is selected', async () => {
+        render(
+          <form data-testid="form">
+            <label htmlFor="iceCreamFlavors">Ice Cream Flavors</label>
+            <Select options={selectOptions} name="iceCreamFlavors" id="iceCreamFlavors" />
+          </form>,
+        );
+        await selectEvent.select(screen.getByLabelText('Ice Cream Flavors'), ['Vanilla']);
+        expect(screen.getByTestId('form')).toHaveFormValues({ iceCreamFlavors: 'vanilla' });
+      });
+    });
+
+    describe('multi select', () => {
+      test('it has no items selected on initial render', () => {
+        render(
+          <form data-testid="form">
+            <label htmlFor="iceCreamFlavors">Ice Cream Flavors</label>
+            <Select options={selectOptions} name="iceCreamFlavors" id="iceCreamFlavors" isMulti />
+          </form>,
+        );
+
+        expect(screen.getByTestId('form')).toHaveFormValues({ iceCreamFlavors: '' });
+      });
+
+      test('it allows the selection of multiple items', async () => {
+        render(
+          <form data-testid="form">
+            <label htmlFor="iceCreamFlavors">Ice Cream Flavors</label>
+            <Select options={selectOptions} name="iceCreamFlavors" id="iceCreamFlavors" isMulti />
+          </form>,
+        );
+        await selectEvent.select(screen.getByLabelText('Ice Cream Flavors'), ['Chocolate', 'Vanilla']);
+        expect(screen.getByTestId('form')).toHaveFormValues({ iceCreamFlavors: ['chocolate', 'vanilla'] });
+
+        await selectEvent.select(screen.getByLabelText('Ice Cream Flavors'), 'Strawberry');
+        expect(screen.getByTestId('form'))
+          .toHaveFormValues({ iceCreamFlavors: ['chocolate', 'vanilla', 'strawberry'] });
+      });
+    });
+  });
```

Comment:
Do we need to be concerned with testing react select itself? If we need to test these scenarios, we should testing our own component.

Reply:
Agreed on this. react-select is technically an implementation detail as far as out users are concerned, so it would make sense to just test our component for these scenarios and let the test fail if for any reason our react-select implementation breaks in the future.

Reply:
Yeah, I wouldn't have done this had it not been for this link being shared. Was there another reason why this was provided? https://testing-library.com/docs/ecosystem-react-select-event